sound/soc/sof/ipc4-topology.c

```diff
@@ -1968,10 +1974,20 @@ static int sof_ipc4_prepare_src_module(struct snd_sof_widget *swidget,
 	int output_format_index, input_format_index;
 	int ret;
 
-	input_format_index = sof_ipc4_init_input_audio_fmt(sdev, swidget, &src->base_config,
-							   pipeline_params, available_fmt);
-	if (input_format_index < 0)
-		return ret;
+	/*
+	 * Initialize the input format and update pipeline memory usage only when the widget is
+	 * prepared for the first time. Otherwise, just update the pipeline params.
+	 */
+	if (!swidget->prepared) {
+		input_format_index = sof_ipc4_init_input_audio_fmt(sdev, swidget,
+								   &src->base_config,
+								   pipeline_params, available_fmt);
+		if (input_format_index < 0)
+			return ret;
+
+		/* update pipeline memory usage */
+		sof_ipc4_update_resource_usage(sdev, swidget, &src->base_config);
```

Comment:
that looks like a different patch?

I also don't get the comment above, where do we do the ''Otherwise, just update the pipeline params"